app/models/issue_tta_report.rb

```diff
@@ -1,5 +1,50 @@
+require "api_delegator"
+
 class IssueTtaReport < ApplicationRecord
-  validates :issue_id, :tta_report_display_id, :tta_report_id, presence: true
-  validates :tta_report_id, format: {with: /\A[0-9]*\z/, message: "only allows numbers"}
+  validates :issue_id, :tta_report_display_id, presence: true
   validates :tta_report_display_id, uniqueness: {scope: :issue_id}
+  validate :api_call_succeeded
+  validates :tta_report_id, presence: true, if: :errors_blank?
+
+  attr_accessor :access_token
+
+  before_validation :retrieve_tta_report_id
```

Comment:
so just to be make sure i understand what's happening here ...
`retrieve_tta_report_id` happens before `.save` runs. then `api_call_succeeded` runs. if there are no errors, than it runs through the validators?

is that the correct passage? this is hard to get at a glance -- can we maybe lump those lines together or in some other way make this a bit more clear?